### PR TITLE
Update prettier dependencies

### PR DESCRIPTION
**Pull request by foxymiles**
### Public-Facing Changes
Fix prettier dependencies so that @foxglove/create-foxglove-extension can create new extensions without errors or warnings.

### Description
Update to latest eslint-plugin-prettier & prettier.

This seems to be a cleaner solution than https://github.com/foxglove/create-foxglove-extension/pull/109 for https://linear.app/foxglove/issue/FG-4317/create-foxglove-extension-not-working.

I tested this with studio and didn't have any issues.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,8 +31,8 @@
     "eslint-plugin-es": "^4",
     "eslint-plugin-filenames": "^1",
     "eslint-plugin-import": "^2",
-    "eslint-plugin-prettier": "^3 || ^4",
```
Comment (defunctzombie) [inline package.json:33]: Do we want to allow `^3 || ^4 || ^5`? I don't remember why we have `^3 || ^4` here in the first place. Arguably we should be specifying a major version we expect for our eslint-plugin? If we don't have a prettier requirement can we remove it altogether?

File: package.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
     "lint": "npm run lint:base && npm run lint:jest && npm run lint:react",
     "lint:base": "eslint --report-unused-disable-directives .",
     "lint:jest": "eslint --report-unused-disable-directives --config ./configs/jest.js .",
-    "lint:react": "eslint --report-unused-disable-directives --config ./configs/react.js ."
+    "lint:react": "eslint --report-unused-disable-directives --config ./configs/react.js .",
+    "prettier": "prettier"
```
Comment (defunctzombie) [inline package.json:21]: Why this change? This is not called out in the PR.

File: package.json [hunk 3]
```diff
@@ -45,5 +46,8 @@
     "eslint-plugin-react-hooks": "4.6.0",
     "jest": "29.5.0",
     "ts-jest": "29.1.0"
+  },
+  "prettier": {
+    "trailingComma": "es5"
```
Comment (defunctzombie) [inline package.json:51]: Why this change? Shouldn't this be in the prettier config?

**Review by defunctzombie — APPROVED** [review]
LGTM but I don't remember why we allowed two major versions in the first place and why that would cause an issue with installation. Maybe @jtbandes remembers?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@foxglove/eslint-plugin",
-  "version": "0.22.1",
+  "version": "0.23.0",
```
Comment (defunctzombie) [inline package.json:3]: We typically do the version bump separately but I think for low-iteration repos like this one it might be alright.

File: package.json (hunk 3 above)
Comment (foxymiles) [reply package.json:51]: What prettier config? We can put this in a separate prettierrc if we want.

This is necessary because the default has changed from `es5` to `all` in the new prettier. Alternatively we can go change the other files in this repo that are affected to the new comma rules.

File: package.json (hunk 2 above)
Comment (foxymiles) [reply package.json:21]: To run the prettier version in the repo from the command line for dev purposes.

File: package.json (hunk 1 above)
Comment (foxymiles) [reply package.json:33]: I'm not sure either. I haven't worked in this repo before today.

File: package.json (hunk 4 above)
Comment (foxymiles) [reply package.json:3]: You'd prefer a separate PR just to bump the version?

Comment (defunctzombie) [reply package.json:3]: For this package I think not - but typically yes cause we might have other PRs we want to merge so no single PR gets to bump versions. Its nicer to feel comfortable reviewing a PR and separately making a decision about which version it will be.

File: package.json (hunk 3 above)
Comment (foxymiles) [reply package.json:51]: For reference:

https://prettier.io/docs/en/options.html#trailing-commas

**Comment by jtbandes** [comment]
git-blame has the history: https://github.com/foxglove/eslint-plugin/pull/36 Not sure if there was a strong reason behind it though.

**Comment by foxymiles** [comment]
> git-blame has the history: #36 Not sure if there was a strong reason behind it though.

Since this was two years ago I'm going to assume we can drop the version 3 support here.
